config/initializers/content_security_policy.rb

```diff
@@ -1,33 +1,33 @@
 # Be sure to restart your server when you modify this file.
-
 # Define an application-wide content security policy
 # For further information see the following documentation
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
 
-if Rails.configuration.csplog_enabled 
+if Rails.configuration.csplog_enabled
   # Report CSP violations to a specified URI
   Rails.application.config.content_security_policy_report_only = true
 
   Rails.application.config.content_security_policy do |policy|
-  ##  Enable all directives supported by Rails 5.2
-    policy.base_uri :self, :https
-    policy.child_src :self, :https
-    policy.connect_src :self, :https
-    policy.default_src :self, :https, :http
-    policy.font_src :self, :https, :http, :data
-    policy.form_action :self, :https
-    policy.frame_ancestors :self, :https
-    policy.frame_src :self, :https
-    policy.img_src :self, :https, :http
-    policy.manifest_src :self, :https, :http
-    policy.media_src :self, :https
+    ##  Enable all directives supported by Rails 5.2
+    policy.base_uri 'self'
+    policy.child_src 'self'
+    policy.connect_src 'self', '*.earthdata.nasa.gov', '*.sit.earthdata.nasa.gov', '*.uat.earthdata.nasa.gov'
+    policy.default_src 'self', '*.earthdata.nasa.gov', '*.sit.earthdata.nasa.gov', '*.uat.earthdata.nasa.gov', 'https://fonts.googleapis.com'
+    policy.font_src 'self', '*.earthdata.nasa.gov', '*.sit.earthdata.nasa.gov', '*.uat.earthdata.nasa.gov', 'https://fonts.googleapis.com'
+    policy.form_action 'self'
+    policy.frame_ancestors 'self'
+    policy.frame_src 'self'
+    policy.img_src 'self', '*.earthdata.nasa.gov', '*.sit.earthdata.nasa.gov', '*.uat.earthdata.nasa.gov', 'https://fonts.googleapis.com'
+    policy.manifest_src 'self'
+    policy.media_src 'self'
     policy.object_src :none
-    policy.script_src :self, :https, :http
-    policy.style_src :self, :https, :http
-    policy.worker_src :self, :https
-    
+    policy.script_src 'self'
+    policy.script_src 'self', '*.earthdata.nasa.gov', '*.sit.earthdata.nasa.gov', '*.uat.earthdata.nasa.gov'
```

Comment:
I'm generally ok with everything, just waiting to see if we can/need to be more specific here

Reply:
script violations come from:

- https://cdn.earthdata.nasa.gov
- https://cdn.sit.earthdata.nasa.gov
- https://fbm.earthdata.nasa.gov

Reply:
the last one took me a few min, but I believe it's the feedback module.

So should we specify those sources, or should we still keep it a little more open? If we make it more closed, I'm wondering how much extra testing we should do and how to know what areas of the application.

Reply:
I prefer to specify those sources since we are only logging violations, not enforcing them. I think we should be on the path toward enforcement, and this moves us in that direction.